opencxl/cxl/transport/transaction.py

```diff
@@ -646,87 +614,110 @@ def get_transaction_id(self) -> int:
         return self.cpl_header.get_transaction_id()
 
 
-class CxlIoCompletionWithDataPacket(CxlIoBasePacket):
-    cpl_header: CxlIoCompletionHeader
-    id: int
-    data: int
-
-    # TODO: Support dynamic data size. Fixed to 8 for now.
-    _fields = CxlIoBasePacket._fields + [
-        StructureField(
-            "cpl_header",
-            CXL_IO_CPL_HEADER_START,
-            CXL_IO_CPL_HEADER_END,
-            CxlIoCompletionHeader,
-        ),
-        DynamicByteField("data", CXL_IO_CPL_FIELD_START, 0x0),
+class CxlIoCplDataPacket(CxlIoBasePacket):
+    cpl_header: CxlIoCplHeader
+    _pack_ = 1
+    _fields_ = [
+        ("cpl_header", CxlIoCplHeader),
     ]
 
     @staticmethod
     def create(
         req_id: int,
         tag: int,
         data: int,
+        pload_len: int,
         status: CXL_IO_CPL_STATUS = CXL_IO_CPL_STATUS.SC,
-        pload_len=0x04,
-    ) -> "CxlIoCompletionWithDataPacket":
-        # for config reads, always 1 DWORD (4 bytes)
+    ):
+        if pload_len == 4:
```

Comment:
We cannot assume fixed data payload size, especially when we start using DMA requests from device. The CPLD TLP should support dynamic data payload.